bufferevent_sock.c

```diff
@@ -431,16 +431,11 @@ bufferevent_socket_connect(struct bufferevent *bev,
 			result = 0;
 			goto done;
 		}
-	} else if (r == 1) {
+	} else {
 		/* The connect succeeded already. How very BSD of it. */
 		result = 0;
 		bufev_p->connecting = 1;
 		bufferevent_trigger_nolock_(bev, EV_WRITE, BEV_OPT_DEFER_CALLBACKS);
-	} else {
-		/* The connect failed already.  How very BSD of it. */
```

Comment:
This code actually wasn't "unreachable" - https://github.com/libevent/libevent/pull/1649